src/formatDate.js

```diff
@@ -9,6 +9,38 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const dateArray = date.split(fromFormat[3]);
+  const dateFromObj = {
+    [fromFormat[0]]: dateArray[0],
+    [fromFormat[1]]: dateArray[1],
+    [fromFormat[2]]: dateArray[2],
```

Comment:
The current implementation assumes that the date parts will always be in the same order as in the `fromFormat` array. This can lead to incorrect behavior if the `date` string is not in the expected order. You should validate the order of the date parts in the `date` string against the `fromFormat` array.